verification/cbmc/include/core_json_annex.h

```diff
@@ -67,45 +67,71 @@
  * Please see core_json.c for documentation.
  */
 
-void skipSpace( const char * buf,
-                size_t * start,
-                size_t max );
-
-bool skipUTF8( const char * buf,
-               size_t * start,
-               size_t max );
-
-bool skipEscape( const char * buf,
-                 size_t * start,
-                 size_t max );
-
-bool skipString( const char * buf,
-                 size_t * start,
-                 size_t max );
-
-bool skipAnyLiteral( const char * buf,
-                     size_t * start,
-                     size_t max );
-
-bool skipDigits( const char * buf,
-                 size_t * start,
-                 size_t max,
-                 int32_t * outValue );
+/* *INDENT-OFF* */
+JSONStatus_t JSON_Validate( const char * buf,
+                            size_t max )
+__CPROVER_requires( max < CBMC_MAX_BUFSIZE )
+__CPROVER_requires( buf == NULL || __CPROVER_is_fresh( buf, max ) )
+__CPROVER_ensures( jsonValidateEnum( __CPROVER_return_value ) );
 
-bool skipNumber( const char * buf,
-                 size_t * start,
-                 size_t max );
-
-bool skipSpaceAndComma( const char * buf,
-                        size_t * start,
-                        size_t max );
+JSONStatus_t skipCollection( const char * buf,
+                             size_t * start,
+                             size_t max )
+__CPROVER_requires( max > 0 && max < CBMC_MAX_BUFSIZE )
```

Comment:
Nitpick: I prefer "0 < max && max < CBMC_MAX_BUFFSIZE" because it is closer to "0 < max < CBMC_MAX_BUFSIZE"